tests/unit/cluster.tcl

```diff
@@ -0,0 +1,97 @@
+# Primitive tests on cluster-enabled redis using redis-cli
+
+source tests/support/cli.tcl
+
+# make sure the test infra won't use SELECT
+set ::singledb 1
+
+set conf {cluster-enabled yes}
+
+# start three servers
+start_server {overrides {cluster-enabled yes} tags {"cluster"}} {
+start_server {overrides {cluster-enabled yes} tags {"cluster"}} {
+start_server {overrides {cluster-enabled yes} tags {"cluster"}} {
+
+    set node1 [srv 0 client]
+    set node2 [srv -1 client]
+    set node3 [srv -2 client]
+
+    test {Create 3 node cluster} {
+        exec src/redis-cli --cluster-yes --cluster create \
+                           127.0.0.1:[srv 0 port] \
+                           127.0.0.1:[srv -1 port] \
+                           127.0.0.1:[srv -2 port]
+    }
+
+    test "Run blocking command on cluster node3" {
+        # key9184688 is mapped to slot 10923 (first slot of node 3)
+        # use DEL command to wait for the cluster to be stable
+        wait_for_condition 1000 50 {
+            ![catch {$node3 del key9184688}]
+        } else {
+            fail "Cluster doesn't stabilize"
+        }
+
```

Comment:
@oranagra what we need is something like this
`test "Cluster is up" {
    assert_cluster_state ok
}`
The following code does not check the cluster state. So we may get CLUSTER DOWN even the check is passed.
`  wait_for_condition 1000 50 {
                [catch {exec src/redis-cli --cluster \
                    check 127.0.0.1:[srv 0 port] \
                }] == 0
            } else {
                fail "Cluster doesn't stabilize"
            }`

Reply:
thanks, implemented.. (with lots of duplicate code that can some day be shared)